### PR TITLE
Remove IncrementalTaskInputs from code

### PR DESCRIPTION
**Pull request by asodja**

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test QFL

**Comment by asodja** [comment]
It seems it's not ready for review yet...

**Comment by lptr** [comment]
If I do a full-text search for `IncrementalTaskInputs` I still have a number of hits that shouldn't be there. For example, this note makes no sense anymore I think:

https://github.com/gradle/gradle/blame/fb22fc394056375873c99e3b4ad3a57925c24771/subprojects/docs/src/docs/userguide/extending-gradle/custom_tasks.adoc#L142-L151

We should also remove `InputChangeTrackingStrategy.ALL_PARAMETERS` and all related functionality.

We should mention the removal in `upgrading_version_7.adoc`, too, right?

I'm guessing we can simplify the code here, too:

https://github.com/gradle/gradle/blob/762c7d1e4f88f89fae0ac27119de381fcbf9ecf3/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompilationSourceDirs.java#L99-L107

I see a lot of tests, especially smoke tests, still expecting the old deprecation warning. I guess it means we need to drop support for the versions producing that warning.

**Comment by asodja** [comment]
@bot-gradle test QFL

**Comment by asodja** [comment]
> I'm guessing we can simplify the code here, too:
> 
> https://github.com/gradle/gradle/blob/762c7d1e4f88f89fae0ac27119de381fcbf9ecf3/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompilationSourceDirs.java#L99-L107
> 

It looks like changing this results in a failure of: `JavaSourceCliIncrementalCompilationIntegrationTest.can remove source root`.

**Comment by asodja** [comment]
@bot-gradle test QFL

**Comment by asodja** [comment]
@bot-gradle test AST

File: subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProviderTest.kt [hunk 1]
```diff
@@ -44,24 +44,22 @@ class EmbeddedKotlinProviderTest : AbstractKotlinIntegrationTest() {
     fun `stdlib and reflect are pinned to the embedded kotlin version for requested plugins`() {
         withBuildScript(
             """
+            buildscript {
+                $repositoriesBlock
+                dependencies {
+                    classpath("org.jetbrains.kotlin:kotlin-stdlib:1.6.10")
```
Comment (wolfs) [inline subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProviderTest.kt:50]: Why are these extra dependencies necessary now? The only thing you need here is the plugin repositories, which could go to `Settings.pluginManagement`, right?

File: subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java [hunk 2]
```diff
@@ -106,10 +106,6 @@ private static IncrementalInputProperties createIncrementalInputProperties(UnitO
         switch (inputChangeTrackingStrategy) {
             case NONE:
                 return IncrementalInputProperties.NONE;
-            //noinspection deprecation
-            case ALL_PARAMETERS:
-                // When using IncrementalTaskInputs, keep the old behaviour of all file inputs being incremental
-                return IncrementalInputProperties.ALL;
```
Comment (wolfs) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java:112]: `IncrementalInputProperties.ALL` should also be removed.

**Review by wolfs — COMMENTED** [review]
Looks good! I left some comments about even more cleanups.

File: subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskExecution.java [hunk 3]
```diff
@@ -423,9 +422,6 @@ public InputChangeTrackingStrategy getInputChangeTrackingStrategy() {
             if (taskAction instanceof IncrementalInputsTaskAction) {
                 return InputChangeTrackingStrategy.INCREMENTAL_PARAMETERS;
             }
-            if (taskAction instanceof IncrementalTaskInputsTaskAction) {
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskExecution.java:426]: We should also merge `AbstractIncrementalTaskAction` with `IncrementalInputsTaskAction`.

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
@bot-gradle test AST

File: subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProviderTest.kt (hunk 1 above)
Comment (asodja) [reply subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProviderTest.kt:50]: This was added because with plugin 1.6.10 `kotlin-stdlib` and `kotlin-reflect` are not seen on `classpath` anymore. I will double check with `kotlin-dsl` experts if that is a good fix.

Comment (asodja) [reply subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProviderTest.kt:50]: @bamboo do you maybe know if `stdlib` and `reflect` are removed from that classpath on our side or there is some other reason why they are not on `classpath` after upgrade to 1.6.10?

File: .teamcity/test-buckets.json [hunk 4]
```diff
@@ -6500,7 +6500,6 @@
 					"org.gradle.api.tasks.CachedLineEndingSensitivityIntegrationSpec=integTest",
 					"org.gradle.api.tasks.DeferredTaskFailureIntegrationTest=integTest",
 					"org.gradle.api.tasks.NestedInputIntegrationTest=integTest",
-					"org.gradle.api.tasks.IncrementalTaskInputsIntegrationTest=integTest",
```
[the remark this answers is not in the document]
Comment (asodja) [reply .teamcity/test-buckets.json]: I reverted that change. So automatic changes will be applied.

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/state/TaskTypeUpToDateIntegrationTest.groovy [hunk 5]
```diff
@@ -16,9 +16,8 @@
 
 package org.gradle.api.internal.changedetection.state
 
-import org.gradle.api.tasks.incremental.IncrementalTaskInputs
+
```
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/state/TaskTypeUpToDateIntegrationTest.groovy]: done

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTaskWithNormalizedInputsIntegrationTest.groovy [hunk 6]
```diff
@@ -16,7 +16,7 @@
 
 package org.gradle.api.tasks
 
-import org.gradle.api.tasks.incremental.IncrementalTaskInputs
+
```
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTaskWithNormalizedInputsIntegrationTest.groovy]: Should be removed.

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTaskWithNormalizedInputsIntegrationTest.groovy [hunk 7]
```diff
@@ -74,17 +69,13 @@ class IncrementalTaskWithNormalizedInputsIntegrationTest extends AbstractIntegra
...
     }
 
     @Issue("https://github.com/gradle/gradle/issues/9320")
-    def "incremental task with NAME_ONLY inputs (matching file names and content) detects deleted file (type: #taskChangeType.simpleName)"() {
+    def "incremental task with NAME_ONLY inputs (matching file names and content) detects deleted file"() {
         def inputs = folderNames().collect { file("${it}/input.txt").createFile() }
 
-        buildFile << incrementalTaskWithNameOnlyInputFiles(taskChangeType, inputs)
-        expectDeprecationWarningForIncrementalTaskInputs(taskChangeType)
+        buildFile << incrementalTaskWithNameOnlyInputFiles(InputChanges, inputs)
```
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTaskWithNormalizedInputsIntegrationTest.groovy]: Removed.

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/UntrackedTaskIntegrationTest.groovy [hunk 8]
```diff
@@ -16,7 +16,7 @@
 
 package org.gradle.api.tasks
 
-import org.gradle.api.tasks.incremental.IncrementalTaskInputs
+
```
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/api/tasks/UntrackedTaskIntegrationTest.groovy]: Should be removed.

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/UntrackedTaskIntegrationTest.groovy [hunk 9]
```diff
@@ -63,7 +63,7 @@ class UntrackedTaskIntegrationTest extends AbstractIntegrationSpec implements Di
 
     def "fails when incremental task is marked as untracked"() {
         file("input/input.txt") << "Content"
-        buildFile(generateUntrackedIncrementalConsumerTask(inputChangesType))
+        buildFile(generateUntrackedIncrementalConsumerTask(InputChanges))
```
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/api/tasks/UntrackedTaskIntegrationTest.groovy]: Removed

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTaskInputsIntegrationTest.groovy [hunk 10]
```diff
@@ -1,134 +0,0 @@
...
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.api.tasks
-
-import org.gradle.internal.execution.history.changes.ChangeTypeInternal
-import spock.lang.Issue
-
-class IncrementalTaskInputsIntegrationTest extends AbstractIncrementalTasksIntegrationTest {
```
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTaskInputsIntegrationTest.groovy:22]: merged

File: subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java (hunk 2 above)
Comment (asodja) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java:112]: Removed

File: subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskExecution.java (hunk 3 above)
Comment (asodja) [reply subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskExecution.java:426]: Merged. Also merged `AbstractIncrementalTaskActionFactory` with `IncrementalInputsTaskActionFactory`

**Comment by asodja** [comment]
@bot-gradle test QFL

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle help

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test APT

**Comment by asodja** [comment]
@bot-gradle test APT

**Comment by asodja** [comment]
@bot-gradle test this

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/UseJdk11OrLaterSampleModifier.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+import org.gradle.util.internal.GUtil;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Modifies runtime Java for samples that require Java11 or later if possible
+ */
+public class UseJdk11OrLaterSampleModifier implements SampleModifier {
```
Comment (wolfs) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/UseJdk11OrLaterSampleModifier.java:34]: We should change these two locations instead: https://github.com/gradle/gradle/blob/2c6b35a6ee248ba9c04b3ee90a3c2974798ed8f0/subprojects/docs/build.gradle#L637-L640
https://github.com/gradle/gradle/blob/2c6b35a6ee248ba9c04b3ee90a3c2974798ed8f0/subprojects/docs/build.gradle#L612-L616

File: subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy [hunk 12]
```diff
@@ -166,12 +166,16 @@ trait ValidationMessageChecker {
     )
     String incorrectUseOfInputAnnotation(@DelegatesTo(value = IncorrectUseOfInputAnnotation, strategy = Closure.DELEGATE_FIRST) Closure<?> spec = {}) {
         def config = display(IncorrectUseOfInputAnnotation, 'incorrect_use_of_input_annotation', spec)
-        config.description("has @Input annotation used on property of type '${config.propertyType}'")
+        def incorrectUseOfInputAnnotation = config.description("has @Input annotation used on property of type '${config.propertyType}'")
```
Comment (wolfs) [inline subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy:169]: Why is this change here?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AndroidPerformanceTestFixture.groovy [hunk 13]
```diff
@@ -29,4 +33,25 @@ trait AndroidPerformanceTestFixture {
         runner.assumeShouldRun()
         AndroidTestProject.projectFor(runner.testProject)
     }
+
+    void configureProjectJavaHomeToJdk11() {
```
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AndroidPerformanceTestFixture.groovy:37]: I don't like changing the Java home for the performance test. Could we run the performance test directly on Java 11? We don't necessarily need to do this as part of this PR.

In general, I think it is time to switch to using Java 11 for performance tests.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy [hunk 14]
```diff
@@ -47,7 +48,8 @@ import static org.gradle.performance.results.OperatingSystem.WINDOWS
 )
 class FasterIncrementalAndroidBuildsPerformanceTest extends AbstractCrossBuildPerformanceTest {
```
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy:49]: We can also remove this test now.

**Review by nathan-contino — APPROVED** [review]
Docs LGTM

File: subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/IncrementalInputsTaskAction.java [hunk 15]
```diff
@@ -17,18 +17,31 @@
...
 
 import org.gradle.api.Task;
+import org.gradle.internal.execution.history.changes.InputChangesInternal;
 import org.gradle.internal.reflect.JavaMethod;
 import org.gradle.work.InputChanges;
 
 import java.lang.reflect.Method;
 
-public class IncrementalInputsTaskAction extends AbstractIncrementalTaskAction {
+public class IncrementalInputsTaskAction extends StandardTaskAction {
```
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/IncrementalInputsTaskAction.java:26]: Sounds better yes

File: subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy (hunk 12 above)
Comment (asodja) [reply subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy:169]: It's because for Directory and DirectoryProperty we have different suggestions in case of incorrect usage on `@Input`. Do you think we should create a different method due to that?

Comment (wolfs) [reply subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy:169]: I think it makes sense to have different suggestions. What I don't understand why this change is in the current PR.

Comment (asodja) [reply subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy:169]: I see. I used this method for catching Sentry Android plugin issues with incorrect inputs. I upgraded plugins in [this commit](https://github.com/gradle/gradle/pull/21481/commits/39b9477c9ee16f64cb1d56230c0180ee69027a0f), since some were not compatible with AGP 7.3+ and after merging the latest changes from master this issue popped up.

Handling of Sentry plugin errors that contain also this change were made in [this commit](https://github.com/gradle/gradle/pull/21481/commits/8908fb5a42cf89f351cf1a5f0fb212b305b62c4d).

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy (hunk 14 above)
Comment (asodja) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/FasterIncrementalAndroidBuildsPerformanceTest.groovy]: Removed

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AndroidPerformanceTestFixture.groovy (hunk 13 above)
Comment (asodja) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AndroidPerformanceTestFixture.groovy:37]: Ok, I will open an issue to fix that in a separate PR.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/UseJdk11OrLaterSampleModifier.java (hunk 11 above)
Comment (asodja) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/UseJdk11OrLaterSampleModifier.java]: Removed UseJdk11OrLaterSampleModifier and used exclussions

File: subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/IncrementalInputsTaskAction.java (hunk 15 above)
[the remark this answers is not in the document]
Comment (asodja) [reply subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/IncrementalInputsTaskAction.java]: renamed

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AndroidPerformanceTestFixture.groovy (hunk 13 above)
Comment (asodja) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AndroidPerformanceTestFixture.groovy:37]: https://github.com/gradle/gradle/issues/21646

**Comment by asodja** [comment]
@bot-gradle test AST

File: subprojects/core/src/main/resources/org/gradle/initialization/removed-types.txt [hunk 16]
```diff
@@ -19,6 +19,10 @@ org.gradle.api.internal.tasks.testing.junit.report.TestReporter
 org.gradle.api.internal.tasks.execution.TaskValidator
 org.gradle.api.artifacts.transform.ArtifactTransformDependencies
 org.gradle.internal.filewatch.FileWatcherFactory
+<<<<<<< HEAD
+org.gradle.api.tasks.incremental.IncrementalTaskInputs
+=======
 org.gradle.api.artifacts.transform.ArtifactTransform
 org.gradle.api.artifacts.transform.ArtifactTransformException
 org.gradle.api.artifacts.transform.VariantTransform
+>>>>>>> master
```
Comment (wolfs) [inline subprojects/core/src/main/resources/org/gradle/initialization/removed-types.txt:28]: You need to remove the conflict markers heres.

**Review by wolfs — COMMENTED** [review]
Nearly there.

File: subprojects/core/src/main/resources/org/gradle/initialization/removed-types.txt (hunk 16 above)
Comment (asodja) [reply subprojects/core/src/main/resources/org/gradle/initialization/removed-types.txt]: Good catch

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test and merge

**Comment by asodja** [comment]
@bot-gradle test AST

File: subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProviderTest.kt (hunk 1 above)
Comment (bamboo) [reply subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/support/EmbeddedKotlinProviderTest.kt:50]: The new `kotlin("jvm")` plugin no longer depends kotlin-stdlib and kotlin-reflect. The change to the test looks good 👍

**Comment by asodja** [comment]
@bot-gradle test and merge
